### ISSUE TITLE
[🐛❗ in esl-event-listener]: performance gap in esl-event-listener auto subscription 

### ISSUE THREAD

**Issue by ala-n** [exadel-inc/esl#1428]
The ESL Event Listener API provides `ESLEventUtils.descriptors` collector to get all the descriptors properties from the object created using the `@listen` annotation. The API is used by `ESLMixinElement` and `ESLBaseElement` as well as it's publically accessible to collect all the descriptors that should provide auto-subscribed ESLEventListener subscriptions.

The current `ESLEventUtils.descriptors` implementation causes a performance issue (reflow and low time of execution) due to `extractValues` utility implementation. 

**Current behavior**
`ESLEventUtils.descriptors` visits all of the enumerable properties that exist on the object, which causes:
 - reflow (due to DOM API getters handling (e.g. `offsetWeight`, `offsetHeight`))
 - causes early initialization of lazy getters as well as accessing heavy getters

**Expected behavior**
`ESLEventUtils.descriptors` should execute fast and do not leas to reflow.

**Additional context**
Basically, there is a 'small' in @listenr implementation as it produces an enumerable property. Potentially that should be handled differently.

**Possible Solution**
Make descriptors auto-collection through the middleware storage (hidden list, or any other similar way) instead of passing through the whole list of object keys and values.
